samcli/commands/remote_invoke/invoke/cli.py

```diff
@@ -0,0 +1,146 @@
+@click.option("--stack-name", required=False, help="Name of the stack to get the resource information from")
+@click.option("--resource-id", required=False, help="Name of the resource that will be invoked")
+@click.option(
+    "--event",
+    "-e",
+    help="The event that will be sent to the resource. The target parameter will depend on the resource type. "
+    "For instance: 'Payload' for Lambda",
+)
+@click.option(
+    "--event-file",
+    type=click.File("r", encoding="utf-8"),
+    help="The file that contains the event that will be sent to the resource",
+)
+@click.option(
+    "--output-format",
+    help="Output format for the boto API response",
+    default="default",
+    type=click.Choice(["default", "full-boto-response"]),
+)
+@remote_invoke_parameter_option
+@event_and_event_file_options_validation
+@common_options
+@aws_creds_options
+@pass_context
+@track_command
+@check_newer_version
+@print_cmdline_args
+def cli(
+    ctx: Context,
+    stack_name: str,
+    resource_id: str,
+    event: str,
+    event_file: TextIOWrapper,
+    output_format: str,
+    parameter: dict,
+    config_file: str,
+    config_env: str,
+) -> None:
+    """
+    `sam cloud invoke` command entry point
+    """
+
+    do_cli(
+        stack_name,
+        resource_id,
+        event,
+        event_file,
+        output_format,
+        parameter,
+        ctx.region,
+        ctx.profile,
+        config_file,
+        config_env,
+    )
+
+
+def do_cli(
+    stack_name: str,
+    resource_id: str,
+    event: str,
+    event_file: TextIOWrapper,
+    output_format: str,
+    parameter: dict,
+    region: str,
+    profile: str,
+    config_file: str,
+    config_env: str,
+) -> None:
+    """
+    Implementation of the ``cli`` method
+    """
+    from samcli.commands.exceptions import UserException
+    from samcli.commands.remote_invoke.remote_invoke_context import RemoteInvokeContext
+    from samcli.lib.remote_invoke.exceptions import (
+        ErrorBotoApiCallException,
+        InvalideBotoResponseException,
+        InvalidResourceBotoParameterException,
+    )
+    from samcli.lib.remote_invoke.remote_invoke_executors import RemoteInvokeExecutionInfo, RemoteInvokeOutputFormat
+    from samcli.lib.utils.boto_utils import get_boto_client_provider_with_config, get_boto_resource_provider_with_config
+
+    boto_client_provider = get_boto_client_provider_with_config(region_name=region)
+    boto_resource_provider = get_boto_resource_provider_with_config(region_name=region)
+
+    try:
+        with RemoteInvokeContext(
+            boto_client_provider=boto_client_provider,
+            boto_resource_provider=boto_resource_provider,
+            stack_name=stack_name,
+            resource_id=resource_id,
+        ) as remote_invoke_context:
+
+            if not resource_id and not stack_name:
+                LOG.error("Atleast 1 of --stack-name or --resource-id parameters should be provided")
+                return
```

Comment:
Let's validate this via click option configuration so we don't need to manually check here.